sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSource.scala

```diff
@@ -81,6 +81,11 @@ class StateDataSource extends TableProvider with DataSourceRegister with Logging
     // The key state encoder spec should be available for all operators except stream-stream joins
     val keyStateEncoderSpec = if (stateStoreReaderInfo.keyStateEncoderSpecOpt.isDefined) {
       stateStoreReaderInfo.keyStateEncoderSpecOpt.get
+    } else if (isReadAllColFamiliesOnJoinV3(sourceOptions)) {
```

Comment:
If you set for joinv3 in `getStoreMetadataAndRunChecks` the way i mentioned in the comment below. Then you wouldn't need this.